src/java.base/unix/classes/sun/nio/fs/UnixPath.java

```diff
@@ -706,53 +708,27 @@ public boolean endsWith(Path other) {
         // compare bytes
         int thisPos = offsets[thisOffsetCount - thatOffsetCount];
         int thatPos = that.offsets[0];
-        if ((thatLen - thatPos) != (thisLen - thisPos))
-            return false;
-        while (thatPos < thatLen) {
-            if (this.path[thisPos++] != that.path[thatPos++])
-                return false;
-        }
-
-        return true;
+        return Arrays.equals(this.path, thisPos, thisLen, that.path, thatPos,
+                thatLen);
     }
 
     @Override
     public int compareTo(Path other) {
-        int len1 = path.length;
-        int len2 = ((UnixPath) other).path.length;
-
-        int n = Math.min(len1, len2);
-        byte v1[] = path;
-        byte v2[] = ((UnixPath) other).path;
-
-        int k = 0;
-        while (k < n) {
-            int c1 = v1[k] & 0xff;
-            int c2 = v2[k] & 0xff;
-            if (c1 != c2) {
-                return c1 - c2;
-            }
-           k++;
-        }
-        return len1 - len2;
+        return Arrays.compareUnsigned(path, ((UnixPath) other).path);
     }
 
     @Override
     public boolean equals(Object ob) {
-        if (ob instanceof UnixPath path) {
-            return compareTo(path) == 0;
-        }
-        return false;
+        return ob instanceof UnixPath p && compareTo(p) == 0;
     }
 
     @Override
     public int hashCode() {
         // OK if two or more threads compute hash
         int h = hash;
         if (h == 0) {
-            for (int i = 0; i< path.length; i++) {
-                h = 31*h + (path[i] & 0xff);
-            }
+            h = ArraysSupport.vectorizedHashCode(path, 0, path.length, 0,
+                    ArraysSupport.T_BOOLEAN);
```

Comment:
Can you make this `/* unsigned bytes */ ArraysSupport.T_BOOLEAN`, or `ArraysSupport.T_BOOLEAN); // unsigned bytes`,  only because T_BOOLEAN isn't always obvious as use-sites.